src/actions/actions.js

```diff
@@ -216,18 +216,33 @@ export function moveCursorToNextChunk() {
   };
 }
 
-export function evaluateNotebook() {
-  return (dispatch, getState) => {
+export const nonRunnableChunkType = ["md", "css", "raw", ""];
+
+const chunkIsRunnable = chunk =>
+  !nonRunnableChunkType.includes(chunk.chunkType);
+
+const chunkNotSkipped = chunk =>
+  !(
+    chunk.evalFlags.includes("skipRunAll") ||
+    chunk.evalFlags.includes("skiprunall")
+  );
+
+export function evaluateNotebookUsingQueue(evalQueueInstance) {
+  return () => (dispatch, getState) => {
     const { jsmdChunks, kernelState } = getState();
+
     if (kernelState !== "KERNEL_BUSY") dispatch(setKernelState("KERNEL_BUSY"));
+
     jsmdChunks.forEach(chunk => {
-      if (!["md", "css", "raw", ""].includes(chunk.chunkType)) {
-        evalQueue.evaluate(chunk, dispatch);
+      if (chunkIsRunnable(chunk) && chunkNotSkipped(chunk)) {
+        evalQueueInstance.evaluate(chunk);
       }
     });
   };
 }
 
+export const evaluateNotebook = evaluateNotebookUsingQueue(evalQueue);
```

Comment:
is there a reason to have this actually be a separate function rather than

```javascript

export function evaluateNotebook(evalQueueInstance=evalQueue) {
...
}

```

This would let you still test it the same way without adding this additional layer (which I'm assuming is the point of this refactor)

Reply:
omfg yes that is so much better! it didn't even occur to me to use a default param!